File: runtime/Swift/Sources/Antlr4/ConsoleErrorListener.swift
/// 
/// Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
/// Use of this file is governed by the BSD 3-clause license that
/// can be found in the LICENSE.txt file in the project root.
/// 


/// 
/// 
/// -  Sam Harwell
/// 

public class ConsoleErrorListener: BaseErrorListener {
    /// 
    /// Provides a default instance of _org.antlr.v4.runtime.ConsoleErrorListener_.
    /// 
    public static let INSTANCE: ConsoleErrorListener = ConsoleErrorListener()

    /// 
    /// 
    /// This implementation prints messages to _System#err_ containing the
    /// values of `line`, `charPositionInLine`, and `msg` using
    /// the following format.
    /// 
    /// line __line__:__charPositionInLine__ __msg__
    /// 
    /// 
    override public func syntaxError<T>(_ recognizer: Recognizer<T>,
                                        _ offendingSymbol: AnyObject?,
                                        _ line: Int,
                                        _ charPositionInLine: Int,
                                        _ msg: String,
                                        _ e: AnyObject?
    ) {
        if Parser.ConsoleError {
            errPrint("line \(line):\(charPositionInLine) \(msg)")
        }
    }

}

File: runtime/CSharp/runtime/CSharp/Antlr4.Runtime/ConsoleErrorListener.cs
/* Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */

#if !PORTABLE

using Antlr4.Runtime;
using Antlr4.Runtime.Sharpen;
using System.IO;

namespace Antlr4.Runtime
{
    /// <author>Sam Harwell</author>
    public class ConsoleErrorListener<Symbol> : IAntlrErrorListener<Symbol>
    {
        /// <summary>
        /// Provides a default instance of
        /// <see cref="ConsoleErrorListener{Symbol}"/>
        /// .
        /// </summary>
        public static readonly ConsoleErrorListener<Symbol> Instance = new ConsoleErrorListener<Symbol>();

        /// <summary>
        /// <inheritDoc/>
        /// <p>
        /// This implementation prints messages to
        /// <see cref="System.Console.Error"/>
        /// containing the
        /// values of
        /// <paramref name="line"/>
        /// ,
        /// <paramref name="charPositionInLine"/>
        /// , and
        /// <paramref name="msg"/>
        /// using
        /// the following format.</p>
        /// <pre>
        /// line <em>line</em>:<em>charPositionInLine</em> <em>msg</em>
        /// </pre>
        /// </summary>
        public virtual void SyntaxError(TextWriter output, IRecognizer recognizer, Symbol offendingSymbol, int line, int charPositionInLine, string msg, RecognitionException e)
        {
            output.WriteLine("line " + line + ":" + charPositionInLine + " " + msg);
        }
    }
}

#endif
